frontend/src/libs/components/tooltip/tooltip-wrapper.tsx

```diff
@@ -3,31 +3,92 @@ import 'react-tooltip/dist/react-tooltip.css';
 import { type FC } from 'react';
 import { Tooltip as ReactTooltip } from 'react-tooltip';
 
-import { type DataTooltipId } from '~/libs/enums/enums.js';
+import { type TooltipPosition } from '~/libs/enums/enums.js';
 import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { useCallback, useEffect, useRef } from '~/libs/hooks/hooks.js';
 import { type ValueOf } from '~/libs/types/types.js';
 
 import styles from './styles.module.scss';
 
 type Properties = {
-  id: ValueOf<typeof DataTooltipId>;
+  id: string;
+  isOpen?: boolean;
+  onClose?: () => void;
   shouldHideArrow?: boolean;
   className?: string;
   offset?: number;
+  openOnClick?: boolean;
+  clickable?: boolean;
+  place?: ValueOf<typeof TooltipPosition>;
+  children?: React.ReactNode;
+  anchorElement?: React.ReactNode;
+  isScrollable?: boolean;
 };
 
 const TooltipWrapper: FC<Properties> = ({
   id,
   shouldHideArrow,
   className,
   offset,
-}) => (
-  <ReactTooltip
-    id={id}
-    noArrow={shouldHideArrow}
-    className={getValidClassNames(className, styles.tooltip)}
-    offset={offset}
-  />
-);
+  openOnClick,
+  clickable,
+  isOpen,
+  place,
+  children,
+  anchorElement,
+  onClose,
+  isScrollable,
+}) => {
+  const tooltipReference = useRef<HTMLDivElement>(null);
+
+  const handleClickOutside = useCallback(
+    (event: MouseEvent): void => {
+      if (
+        tooltipReference.current &&
+        !tooltipReference.current.contains(event.target as Node)
+      ) {
+        onClose && onClose();
+      }
+    },
+    [onClose],
+  );
```

Comment:
Currently, there is an issue with disabling scroll. When clicking outside the tooltip and it's a click on another more actions button, two handlers are triggering: click on the button and the outside click. And the state of isOpen will be true, and then false. The tooltip is visible, but in this case, the document scroll isn't disabled as it's related to the isOpen state. Could you suggest some solution here?
Meanwhile, adding a ref to the ReactTooltip itself doesn't work, I added it to the anchorElement, which is a more actions button in this case.

Reply:
I'm not sure if it is a good idea to use `ReactTooltip` for popovers at all because tooltips have been designed for quite different purposes, so it might produce unexpected behavior (like this one) if we use them differently. Maybe we should take a look at some other packages (Radix UI Popover for example). It's a headless composable component, with all accessibility and functionality in place.
@andanf-e what do you think?

https://www.radix-ui.com/primitives/docs/components/popover
https://www.npmjs.com/package/@radix-ui/react-popover

Reply:
```
useEffect(() => {
    if (isOpen && !isScrollable) {
      document.body.classList.add(styles.noOverflow);
    }

    return () => {
      document.body.classList.remove(styles.noOverflow);
    };
  }, [isOpen, isScrollable]);
```
I added this code to the popover component to disable scroll when the popover is opened. But the issue with the scroll is actually the same as it was before. Should I handle it differently or just omit the scroll functionality for now? Also, as there is a popover component, do I need to remove modifications in the tooltip component?

Reply:
Yep, left it as it is with scrolling,
And yes, remove changes from tooltip